daemon/state.go

```diff
@@ -222,6 +222,8 @@ func (d *Daemon) regenerateRestoredEndpoints(state *endpointRestoreState) (resto
 			ctmap.DeleteIfUpgradeNeeded(ep)
 		}
 
+		ep.SetAllocator(d.identityAllocator)
```

Comment:
Is there any chance to move this earlier? This feels like something that should happen when the endpoint object itself is created, rather than later on.

Reply:
No, we need to do the following in the following order:

* restore endpoints, and validate them, including allocating their IPs
* allocate IPs for the node & start node discovery
* initialize the identity allocator. This must be done after the node addressing has been configured (which has to be done after we allocate the IPs for restored endpoints), because we require to have the node IP as a suffix for the key in the key-value store.

See the following code order:

https://github.com/cilium/cilium/blob/72a7c97e91b63217e7f06e865dbd86379d26df42/daemon/daemon.go#L451-L496

Reply:
Oh, I see. Hmm. OK, thanks for the explanation.

Reply:
@ianvernon please add a shorter explanation for this in this line.